### PR TITLE
MTDSA-12856 Prepare common code for the TYS endpoint changes

### PR DESCRIPTION
**Pull request by mattstephens-hmrc**

### CODE REVIEW

File: app/v1/connectors/BaseDownstreamConnector.scala [hunk 1]
```diff
@@ -93,14 +60,14 @@ trait BaseDownstreamConnector {
     doGet(getBackendHeaders(uri, hc, correlationId))
   }
 
-  def getWithQueryParams[Resp](uri: DownstreamUri[Resp], queryParams: Seq[(String, String)])(implicit
+  def get[Resp](uri: DownstreamUri[Resp], queryParams: Seq[(String, String)] = Nil)(implicit
```
Comment (harry-ritchie) [inline app/v1/connectors/BaseDownstreamConnector.scala:63]: Is there a reason we set QPs to Nil here?

File: app/v1/connectors/ListBenefitsConnector.scala [hunk 2]
```diff
@@ -40,9 +40,9 @@ class ListBenefitsConnector @Inject() (val http: HttpClient, val appConfig: AppC
     val nino    = request.nino.nino
     val taxYear = request.taxYear
 
-    getWithQueryParams(
+    get(
       IfsUri[ListBenefitsResponse[HMRCStateBenefit, CustomerStateBenefit]](s"income-tax/income/state-benefits/$nino/$taxYear"),
-      request.benefitId.map("benefitId" -> _).toSeq
+      queryParams = request.benefitId.map("benefitId" -> _).toSeq
```
Comment (harry-ritchie) [inline app/v1/connectors/ListBenefitsConnector.scala:45]: Ah, I am guessing QPs are set to 'Nil' in the BaseDownstreamConnector for the case that we don't provide a benefitId, right? However benefitId is not a QP so does it make this code confusing to read?

Comment (mattstephens-hmrc) [reply app/v1/connectors/ListBenefitsConnector.scala:45]: If you call to Seq on an empty Option (i.e. None) you'll get an empty Seq, basically Nil. benefitId is a query parameter, downstream.

File: test/v1/connectors/BaseDownstreamConnectorSpec.scala [hunk 3]
```diff
@@ -211,7 +211,7 @@ class BaseDownstreamConnectorSpec extends ConnectorSpec {
             .parameterGet(absoluteUrl, params, config, requiredHeaders, excludedHeaders)
             .returns(Future.successful(outcome))
 
-          await(connector.getWithQueryParams(IfsUri[Result](url), params)) shouldBe outcome
```
Comment (harry-ritchie) [inline test/v1/connectors/BaseDownstreamConnectorSpec.scala:214]: I think we need the TysIfs tests here as well

Comment (mattstephens-hmrc) [reply test/v1/connectors/BaseDownstreamConnectorSpec.scala:214]: > I think we need the TysIfs tests here in the BaseDownstreamConnectorSpec

BaseDownstreamConnectorSpec doesn't really need them added, there isn't much value (if any) adding extra tests there

Good catch re TysIfsTest in ConnectorSpec though, that was missing a chunk of TYS code

**Review by harry-ritchie — APPROVED** [review]

**Pull request merged**